packages/astro/src/runtime/server/endpoint.ts

```diff
@@ -40,7 +40,11 @@ ${chosenMethod} requests are not available when building a static site. Update y
 		return response;
 	}
 
-	if (handler.length > 1) {
+	// Suppress warning in CI so it doesn't clutter the logs
+	// TODO: Remove support for old API in Astro 3.0
```

Comment:
Yeah I was a bit worried with the breaking change, since if we remove the warning, it would be good to remove the old behaviour so there's no unexpected old API usage again. IIRC some of our tests are testing the old API too so we'd have to clean that up. I'll see what Matthew has in mind too.

Reply:
I say we hold off on removing until 3.0. I can't imagine that people are actually doing this now and just ignoring the warning... but you never know.

We can go ahead and remove the tests though. Just remove the actual TODO in 3.0

Reply:
Removed the tests in 1e794a4